mobile/examples/phi-3/android/app/src/main/java/ai/onnxruntime/genai/demo/MainActivity.java

```diff
@@ -117,22 +138,34 @@ public void run() {
 
                             generatorParams = model.createGeneratorParams();
                             //examples for optional parameters to format AI response
-                            //generatorParams.setSearchOption("length_penalty", 1000);
-                            //generatorParams.setSearchOption("max_length", 500);
+                            generatorParams.setSearchOption("length_penalty", lengthPenalty);
+                            generatorParams.setSearchOption("max_length", maxLength);
 
                             encodedPrompt = tokenizer.encode(promptQuestion_formatted);
                             generatorParams.setInput(encodedPrompt);
 
                             generator = new Generator(model, generatorParams);
 
+                            // try to measure average time taken to generate each token.
+                            long startTime = System.currentTimeMillis();
+                            long currentTime = startTime;
+                            int numTokens = 0;
                             while (!generator.isDone()) {
                                 generator.computeLogits();
                                 generator.generateNextToken();
                  
                                 int token = generator.getLastTokenInSequence(0);
                  
                                 tokenListener.accept(stream.decode(token));
+
+                                Log.i(TAG, "Generated token: " + token + ": " +  stream.decode(token));
+                                Log.i(TAG, "Time taken to generate token: " + (System.currentTimeMillis() - currentTime)/ 1000.0 + " seconds");
+                                currentTime = System.currentTimeMillis();
+                                numTokens++;
                             }
+                            long totalTime = System.currentTimeMillis() - startTime;
+                            Log.i(TAG, "Total time taken to generate + " + numTokens + "tokens: " + totalTime);
+                            Log.i(TAG, "Tokens generated per second: " + 1000 * (numTokens / totalTime));
```

Comment:
the time taken to generate the first token (prompt processing) may be significantly longer than the time taken to generate subsequent tokens.

it may be useful to report those measurements separately. e.g., prompt processing time (from duration of first token generation) and token generation time and/or rate (from durations of generating subsequent tokens).